### ISSUE TITLE
feature request: Implement Cognito::UserPoolResourceServer

### ISSUE THREAD

**Issue by horsfallnathan** [localstack/localstack#7161]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

"The AWS::Cognito::UserPoolResourceServer resource creates a new OAuth2.0 resource server and defines custom scopes in it." - [Docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpoolresourceserver.html)

Currently, this has not been implemented in Localstack and it yields this error when you attempt to create one:
``` bash
localstack | 2022-11-10T14:29:10.001 DEBUG --- [   asgi_gw_1] l.u.c.template_deployer    : Extract resource attribute: Cognito::UserPoolResourceServer Ref
localstack | 2022-11-10T14:29:10.001  WARN --- [   asgi_gw_1] l.u.c.template_deployer    : Unexpected resource type Cognito::UserPoolResourceServer when resolving references of resource UserPoolV4ServicetoServiceResourceServer18E67627: {"Type": "AWS::Cognito::UserPoolResourceServer", "Metadata": {"aws:cdk:path": "XXXXX/XXXX-YYYY/UserPool-V4-ResourceServer/Resource"}, "LogicalResourceId": "UserPoolV4ResourceServer1627", "Properties": {"Identifier": "services", "Name": "services", "UserPoolId": "us-east-1_b9d16", "Scopes": [{"ScopeDescription": "Service-to-service scope", "ScopeName": "fullAccess"}]}, "PhysicalResourceId": null}. To find out if Cognito::UserPoolResourceServer is supported in LocalStack Pro, please check out our docs at https://docs.localstack.cloud/aws/cloudformation
```

**Comment by horsfallnathan** [comment]
Hi @calvernaz, thanks for taking a look at this. I don't see any linked PR, is the fix available now?

**Comment by calvernaz** [comment]
Hi @horsfallnathan, authorizers are part of the upstream project, thus the PR not being visible.

**Comment by horsfallnathan** [comment]
Great!! @calvernaz one more thing, is this change available in v1.3, or it ships with a later version?
